spec/message_spec.rb

```diff
@@ -48,9 +48,35 @@ def encode_to_stream(stream)
         stream << Protobuffness::Uint32.encode_attribute(@attributes[:age], 2)
       end
     end
+
+    ## Decoding
+
+    def self.decode(bytestring)
+      stream = StringIO.new(bytestring)
+      decode_from_stream(stream)
+    end
+
+    def self.decode_from_stream(stream)
+      sally = Sally.new({})
+      Protobuffness::Decoder.decode_each_field(stream) do |tag, value|
+        assign_field_to_message(tag, value, sally)
+      end
+      sally
+    end
+
+    def self.assign_field_to_message(tag, value, sally)
+      case tag
+      when 1
+        sally.mood = value
+      when 2
+        sally.age = value
+      else
+        # assign to unknown attributes
```

Comment:
yes, I'm leaving myself a note to come back and finish implementing this. The current `protobuf` gem that we use doesn't implement unknown attributes (I believe), but they are part of the protocol buffers spec